### PR TITLE
Create smart contract for Pentamidine isethionate 

### LINKED ISSUE #18: Create smart contract for Pentamidine isethionate
ℹ️ This issue is part of Epic #14 

## Objective

Create a smart contract to calculate the dosage for Pentamidine isethionate for Leishmaniasis, a disease transmitted by vectors. There are a preparation and treatment time field in the contract response with the following information:

Preparation
- Reconstitute each ampoule with 5 mL of water for injections.
- Take blood glucose and blood pressure before administration.
- Apply the calculated volume IM.
- Concomitantly, administer IV 100 to 200 mL of 5% DAD for 30 min.
- Observe the patient for 1 hour controlling vital signs and glucometry.
- Cite in 48 hours for the next dose.

Treatment time: Cutaneous Leishmaniasis = 4 doses Mucocutaneous Leishmaniasis = 7 to 15 dosis Visceral Leishmaniasis = 7 to 15 doses

## Checklist

- [ ] The smart contract calculates the correct dosage for Pentamidine isethionate
- [ ] The smart contract has been deployed to the testnet
- [ ] There is a preparation field in the response with the text specified
- [ ] There is a unit test for the smart contract

### PR DESCRIPTION
**Pull request by keliumJU**
## Description

The translation keys were added for the Pentamidine Isethionate contract.

Fixes #18

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: okdose/package.json [hunk 1]
```diff
@@ -21,7 +21,8 @@
     "@near-wallet-selector/wallet-connect": "^7.0.0",
     "near-api-js": "^0.44.2",
     "near-cli": "^3.4.0",
-    "near-sdk-js": "0.6.0"
+    "near-sdk-js": "0.6.0",
+    "yarn": "^1.22.19"
```
Comment (einerzg) [inline okdose/package.json:25]: Please remove this line. Install `yarn` global for you!!!

File: okdose/transmissionTypes/transmittedByVectors/leishmaniasis/index.js [hunk 2]
```diff
@@ -6,4 +6,8 @@ const prescribeSodiumStibogluconate = async (nearInterface, weight) => (
   nearInterface.viewMethod(process.env.STIBOGLUCONATE_CONTRACT, 'get_dose', { weight })
 );
 
-export { prescribeNMetilGlucamine, prescribeSodiumStibogluconate };
+const prescribePentamidineIsethionate = async (nearInterface, weight) => (
+  nearInterface.viewMethod(process.env.PENTAMIDINE_CONTRACT, 'get_dose', { weight })
```
Comment (einerzg) [inline okdose/transmissionTypes/transmittedByVectors/leishmaniasis/index.js:10]: `get_dose` should be a const.

File: okdose/transmissionTypes/transmittedByVectors/leishmaniasis/index.js [hunk 3]
```diff
@@ -6,4 +6,8 @@ const prescribeSodiumStibogluconate = async (nearInterface, weight) => (
...
+);
+
+export { prescribeNMetilGlucamine, prescribeSodiumStibogluconate, prescribePentamidineIsethionate };
```
Comment (einerzg) [inline okdose/transmissionTypes/transmittedByVectors/leishmaniasis/index.js:13]: change the format to:

```
export { 
    prescribeNMetilGlucamine,
    prescribeSodiumStibogluconate,
    prescribePentamidineIsethionate
};
```

File: okdoseweb/App.js [hunk 4]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { useTranslation } from 'react-i18next';
 import { NearInterface } from '../okdose/near/near-interface';
-import { prescribeSodiumStibogluconate } from '../okdose/transmissionTypes/transmittedByVectors/leishmaniasis';
+import { prescribePentamidineIsethionate } from '../okdose/transmissionTypes/transmittedByVectors/leishmaniasis';
```
Comment (einerzg) [inline okdoseweb/App.js:5]: this path is too long, you can use this to solve the case:

https://www.npmjs.com/package/babel-plugin-module-resolver

File: okdoseweb/App.js [hunk 5]
```diff
@@ -11,7 +11,7 @@ export default function App({ wallet }) {
   const [valueFromBlockchain, setValueFromBlockchain] = React.useState({});
 
   React.useEffect(() => {
-    prescribeSodiumStibogluconate(nearInterface, 50)
+    prescribePentamidineIsethionate(nearInterface, 80)
```
Comment (einerzg) [inline okdoseweb/App.js:14]: Please, avoid hardcoding code. For this case create a const for 80.

File: okdoseweb/App.js (hunk 4 above)
Comment (einerzg) [reply okdoseweb/App.js:5]: @LauraVilladakommit

**Review by einerzg — DISMISSED** [review]
Some comments @keliumJU.

File: okdose/transmissionTypes/transmittedByVectors/leishmaniasis/index.js (hunk 2 above)
Comment (jcamilocp) [reply okdose/transmissionTypes/transmittedByVectors/leishmaniasis/index.js:10]: We thought of doing that. Maybe we can declare it as a const, as it's the same for almost every contract

File: okdoseweb/App.js (hunk 4 above)
Comment (jcamilocp) [reply okdoseweb/App.js:5]: This file is just being used for tests. It won't be like that for the final version.

File: okdoseweb/App.js (hunk 5 above)
Comment (jcamilocp) [reply okdoseweb/App.js:14]: Just testing again. 80 isn't a meaningful value, just a random one

Comment (LauraVilladakommit) [reply okdoseweb/App.js:14]: @einerzg That file is just to test the call to the contract from the UI, the components will have a weight prop to pass to the prescription.

File: okdose/package.json (hunk 1 above)
Comment (keliumJU) [reply okdose/package.json:25]: I think it was installed by version manager **asdf**, I remove it now

File: okdose/transmissionTypes/transmittedByVectors/leishmaniasis/index.js (hunk 3 above)
Comment (keliumJU) [reply okdose/transmissionTypes/transmittedByVectors/leishmaniasis/index.js:13]: it looks better :)

File: okdoseweb/App.js (hunk 5 above)
Comment (keliumJU) [reply okdoseweb/App.js:14]: I created a weight constant for the test

File: .gitignore [hunk 6]
```diff
@@ -35,3 +35,5 @@ yarn-error.log*
 /okdoseweb/.parcel-cache
 
 /**/.eslintcache
+
+/okdose/contracts.env
```
Comment (jcamilocp) [inline .gitignore:39]: I think we should move the `.env` file to the root folder, as it is mainly used in the front-end

File: okdoseweb/package.json [hunk 7]
```diff
@@ -3,19 +3,23 @@
   "version": "0.1.0",
   "license": "Apache-2.0",
   "scripts": {
-    "start": "env-cmd -f ../okdose/okdose-dapp-contracts/neardev/contracts.env parcel index.html --open",
+    "start": "env-cmd -f ../okdose/contracts.env parcel index.html --open",
```
Comment (jcamilocp) [inline okdoseweb/package.json:6]: If we decide to move the file, we should change this line

**Review by jcamilocp — COMMENTED** [review]
Question: should we update the README, adding the steps to create the `contracts.env` file? @LauraVilladakommit

File: okdoseweb/App.js (hunk 5 above)
Comment (jcamilocp) [reply okdoseweb/App.js]: That's enough for me tbh. @einerzg what do you think?

File: okdoseweb/package-lock.json [hunk 8]
```diff
@@ -0,0 +1,7799 @@
+{
```
Comment (einerzg) [inline okdoseweb/package-lock.json:1]: What will you use `npm` or `yarn`? @keliumJU & @LauraVilladakommit

File: package-lock.json [hunk 9]
```diff
@@ -0,0 +1,8593 @@
+{
```
Comment (einerzg) [inline package-lock.json:1]: What will you use npm or yarn?

Comment (keliumJU) [reply package-lock.json:1]: here too, we will only use yarn :)

**Comment by keliumJU** [comment]
@einerzg the error that presented the linter in the CI was solved, @jcamilocp  new features were added to the linter to control the imports by means of aliases with the module resolver of babel, probably the same process should be done in okdose and okdose-dapp-contract.

File: okdoseweb/package.json [hunk 10]
```diff
@@ -3,25 +3,32 @@
...
     "build": "parcel build index.html --public-url ./"
   },
   "devDependencies": {
     "@babel/core": "^7.18.2",
+    "@babel/eslint-parser": "^7.19.1",
+    "@babel/eslint-plugin": "^7.19.1",
     "@babel/preset-env": "^7.18.2",
     "@babel/preset-react": "^7.17.12",
     "@types/node": "^18.6.2",
+    "add": "^2.0.6",
```
Comment (jcamilocp) [inline okdoseweb/package.json:16]: Remove this, please

File: okdoseweb/package.json [hunk 11]
```diff
@@ -3,25 +3,32 @@
...
     "eslint-plugin-jsx-a11y": "^6.6.1",
+    "eslint-plugin-module-resolver": "^1.5.0",
     "eslint-plugin-react": "^7.31.11",
     "eslint-plugin-react-hooks": "^4.6.0",
     "nodemon": "^2.0.16",
     "parcel": "^2.6.0",
     "process": "^0.11.10",
-    "react-test-renderer": "^18.1.0"
+    "react-test-renderer": "^18.1.0",
+    "yarn": "^1.22.19"
```
Comment (jcamilocp) [inline okdoseweb/package.json:31]: And remove this also

File: okdoseweb/package.json [hunk 12]
```diff
@@ -3,25 +3,32 @@
...
+    "@babel/eslint-parser": "^7.19.1",
+    "@babel/eslint-plugin": "^7.19.1",
     "@babel/preset-env": "^7.18.2",
     "@babel/preset-react": "^7.17.12",
     "@types/node": "^18.6.2",
+    "add": "^2.0.6",
+    "babel-plugin-module-resolver": "^5.0.0",
     "env-cmd": "^10.1.0",
     "eslint": "^8.28.0",
     "eslint-config-airbnb": "^19.0.4",
```
Comment (jcamilocp) [inline okdoseweb/package.json:20]: Should we remove this dependency?

File: okdoseweb/index.js [hunk 13]
```diff
@@ -1,9 +1,9 @@
 // React
```
Comment (jcamilocp) [inline okdoseweb/index.js:1]: Let's remove this line

**Review by jcamilocp — CHANGES_REQUESTED** [review]

File: okdoseweb/package.json (hunk 10 above)
Comment (keliumJU) [reply okdoseweb/package.json:16]: done

File: okdoseweb/package.json (hunk 11 above)
Comment (keliumJU) [reply okdoseweb/package.json:31]: this comes from the main branch, just moved it but done

File: okdoseweb/package.json (hunk 12 above)
Comment (keliumJU) [reply okdoseweb/package.json:20]: not only `add`, `babel-plugin-module-resolver` that was the plugin that @einerzg  recommended.

File: okdoseweb/index.js (hunk 13 above)
Comment (keliumJU) [reply okdoseweb/index.js:1]: Done, and also removed other unnecessary dependencies

**Review by jcamilocp — APPROVED** [review]
LGTM!

**Pull request merged**